src/snowflake/snowpark/_internal/telemetry.py

```diff
@@ -79,6 +81,9 @@ class TelemetryField(Enum):
 }
 APIS_WITH_MULTIPLE_CALLS = list(API_CALLS_TO_ADJUST.keys())
 
+_Args = ParamSpec("_Args")
```

Comment:
ParamSpec is introduced in Python 3.10. We currently support Python3.8+.

Reply:
I saw there's a workaround in `typing_extensions` for python 3.7-3.9, does that work?